PsfRuntime/CreateProcessHook.cpp

```diff
@@ -153,19 +157,68 @@ BOOL WINAPI CreateProcessFixup(
         processInformation = &pi;
     }
 
-    if (!CreateProcessImpl(
-        applicationName,
-        commandLine,
-        processAttributes,
-        threadAttributes,
-        inheritHandles,
-        creationFlags | CREATE_SUSPENDED,
-        environment,
-        currentDirectory,
-        startupInfo,
-        processInformation))
+    auto appConfig = PSFQueryCurrentAppLaunchConfig(true);
+    bool preventBreakAway = false;
+    auto preventBreakAwayPtr = appConfig->try_get("InPackageContext");
+    if (preventBreakAwayPtr)
+    {
+        preventBreakAway = preventBreakAwayPtr->as_boolean().get();
+    }
+
+    if (preventBreakAway)
     {
-        return FALSE;
+        STARTUPINFOEX startupInfoEx =

```

Comment:
Just realized that if preventBreakaway is true this code path ignores StartupInfo.  The data in StartupInfo should be merged with StartupInfoEx.

Reply:
Oh, Yes. Thanks for pointing it out